docs/source/deploying-volttron/proxy_web_server.rst

```diff
@@ -0,0 +1,323 @@
+This will be done for this demo, but is not recommended (see warning and link above)
+
+.. code-block::
+    openssl rsa -in server.key -out server.key.insecure
+    mv server.key server.key.secure
+    mv server.key.insecure server.key
+
+Generate a CSR.
+
+.. code-block::
+    openssl req -new -key server.key -out server.csr
+
+Create the certificate file using the CSR and key files:
+
+.. code-block::
+
+    openssl x509 -req -days 365 -in server.csr -signkey server.key -out server.crt
+
+server.key can now be used for the SSLCertificateKeyFile entry,
+and server.crt can now be used for the SSLCertificateFile entry.
+
+
+ Once that is complete, verify that ProxyPass and ProxyPassReverse options,
+as well as the /vc path found under the RewriteRule option are all appropriate for your use case. This simple use-case
+routes all web traffic through the proxy to your VOLTTRON instance running on your localhost.
+
+
+.. code-block:: console
+
+    <IfModule mod_ssl.c>
+        SSLPassPhraseDialog  exec:/usr/share/apache2/ask-for-passphrase
+        SSLSessionCache		shmcb:${APACHE_RUN_DIR}/ssl_scache(512000)
+        SSLSessionCacheTimeout  300
+        SSLRandomSeed startup file:/dev/urandom  256
+        SSLRandomSeed connect builtin
+        SSLCryptoDevice builtin
+
+        <VirtualHost _default_:443>
+            ErrorLog logs/ssl_error_log
+            TransferLog logs/ssl_access_log
+            LogLevel warn
+            SSLEngine on
+            # Require TLS 1.2 only
+            SSLProtocol -all +TLSv1.2
+            # Require Strong Ciphers
+            SSLCipherSuite "EECDH:!RC4:!3des:!SHA"
+            # Require Cipher Order
+            SSLHonorCipherOrder on
+            SSLCertificateFile <path to cert file provided by CA>
+            SSLCertificateKeyFile <path to key file>
+            <Files ~ "\.(cgi|shtml|phtml|php3?)$">
+                SSLOptions +StdEnvVars
+            </Files>
+            BrowserMatch "MSIE [2-5]" \
+                     nokeepalive ssl-unclean-shutdown \
+                     downgrade-1.0 force-response-1.0
+            CustomLog logs/ssl_request_log \
+                      "%t %h %{SSL_PROTOCOL}x %{SSL_CIPHER}x \"%r\" %b"
+
+            #Setup proxy for volttron
+            ProxyRequests Off
+            ProxyPreserveHost Off
+            ProxyVia Off
+            # Set proxy path as appropriate.
+            ProxyPass          /     http://localhost:8080/ timeout=60
+            ProxyPassReverse   /     http://localhost:8080/ timeout=60
+
+            #Setup websockets to proxy for volttron central
+            RewriteEngine On
+            RewriteCond %{HTTP:UPGRADE} ^WebSocket$ [NC]
+            RewriteCond %{HTTP:CONNECTION} Upgrade$ [NC]
+            # Set proxy path as appropriate
+            RewriteRule /vc/index.html#/dashboard(.*) ws://localhost:8080/vc/index.html#/dashboard$1 [P]
```

Comment:
This is only needed if we have vc installed...

Reply:
Duly noted!